scripts/tuple_tree_generator/tuple_tree_generator/schema/schema.py

```diff
@@ -90,6 +92,38 @@ def _resolve_references(self):
         for t in self.definitions.values():
             t.resolve_references(self)
 
+    def _generate_kinds(self):
+        children = defaultdict(list)
+        for key, value in self.definitions.items():
+            if isinstance(value, StructDefinition) and value._inherits is not None:
+                parent = self.get_definition_for(value._inherits)
+                children[parent].append(value)
+
+        for parent, child_list in children.items():
+            kind_field = [f for f in parent.fields if f.name == "Kind"]
+            if len(kind_field) == 0:
+                raise ValueError("Kind field must be present in abstract classes")
+            if kind_field[0].type != f"{parent.user_namespace}::{parent.name}Kind::Values":
+                raise ValueError(
+                    f"""
+                    Kind field in {parent.name} must have the type
+                    '{parent.user_namespace}::{parent.name}Kind::Values'.
+                    This enum will be autogenerated with the child names
+                    of this class.
+                """
+                )
```

Comment:
I think this message will be printed with indentation.

Reply:
Indeed, fixed